csrc/scheduler/vectorize_helper.cpp

```diff
@@ -920,6 +1019,15 @@ int64_t getVectorizationFactor(
         max_vec_size);
   }
 
+  for (const auto resize_factor : resize_factors) {
+    auto inferred_val =
+        runtime_info.expressionEvaluator().evaluate(resize_factor);
+    if (!inferred_val.hasValue()) {
+      return 1;
+    }
+    max_vec_size = std::gcd(max_vec_size, inferred_val.as<int64_t>());
```

Comment:
One thing I think the WAR is different from how we are handling resize in `propagateRezie` is that: the WAR here doesn't consider the position of the ID where the resize is applying to.

Here we are taking gcd over each factor in the resize op. (left_extent & right_extent & (in|out extent)).
So if the resize is only applying on a middle dimension, I think we are not supposed to take the GCD of that resize on the overall `max_vec_size`.

Reply:
I think the approach taken here is that we are going to be more conservative on applying vectorization with resize.

For rope, since the resize is happening at the fastest dimension, it does seem to be doing the right thing. But this will be restrictive on more general cases.

Reply:
You're right, that's a good point. I'd keep it as is since this is an immediate correctness issue. To make it more precise, it would need more significant changes.

Reply:
Added a comment.

Reply:
If innermost two contiguous IDs looked like `{1024, 6}`, and the innermost ID were sliced as `[2:4]`, would our current analysis yield 4 as its vectorization factor instead of 2? Because of the innermost ID, it should be 2. However, its projected extent would be 2 but then the outer ID has a projected extent of 1024, so if two IDs were merged, wouldn't it (falsely) claim it should be vectorizable by a factor of 4?

Reply:
No. when we resize a ID, we excludes all its outer dimensions from the contiguous IDs. https://github.com/NVIDIA/Fuser/blob/b0bf8f39c9fe64db76983e74eae06aec03c3da5d/csrc/scheduler/vectorize_helper.cpp#L381-L384

Reply:
That sounds fair. Let me take another round at the graph_traversal stuff.